server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/QueryToFilterAdapter.java

```diff
@@ -179,7 +176,11 @@ protected final boolean countCanUseMetadata(FiltersAggregator.Counter counter, B
         BooleanQuery.Builder builder = new BooleanQuery.Builder();
         builder.add(query, BooleanClause.Occur.MUST);
         builder.add(extraQuery, BooleanClause.Occur.MUST);
-        return new QueryToFilterAdapter<>(searcher(), key(), builder.build());
+        return new QueryToFilterAdapter<>(searcher(), key(), builder.build()) {
+            public boolean isInefficientUnion() {
+                return true;
+            }
+        };
```

Comment:
I'm mostly building the boolean query here out of spite - we don't use it but it is the correct query to use if we didn't disable the optimization.